Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -256,14 +268,17 @@ struct PathHierarchy {
                 moduleNode.add(symbolChild: topLevelNode)
             }
             
-            assert(
-                topLevelCandidates.values.filter({ $0.symbol!.pathComponents.count > 1 }).allSatisfy({ $0.parent == nil }), """
-                Top-level candidates shouldn't already exist in the hierarchy. \
-                This wasn't true for \(topLevelCandidates.filter({ $0.value.symbol!.pathComponents.count > 1 && $0.value.parent != nil }).map(\.key).sorted())
-                """
-            )
+            assertAllNodes(in: topLevelCandidates.values.filter { $0.symbol!.pathComponents.count > 1 }, satisfy: { $0.parent == nil },
+                           "Top-level candidates shouldn't already exist in the hierarchy.")
             
             for node in topLevelCandidates.values where node.symbol!.pathComponents.count > 1 && node.parent == nil {
+                symbolNodesOutsideOfHierarchyByModule[moduleNode.symbol!.identifier.precise, default: []].append(node)
+            }
+        }
+        
+        for (moduleID, nodes) in symbolNodesOutsideOfHierarchyByModule {
+            let moduleNode = roots[moduleID]!
```

Comment:
accessing value of roots with key as moduleID without checking roots exits? may be that check already exists somewhere else beyond the change you are proposing here.

Reply:
It's not necessary to check if the module node exists in the roots dictionary. These module nodes are created locally in this initializer and [the only code that creates them](https://github.com/swiftlang/swift-docc/pull/1220/files#diff-e0d0f88acc1bca134d47867a977e60a988a20118c4b52451f90376f6e3d56f7cL90-R106) always adds them to the `roots` dictionary.

Reply:
I see now why it is not necessary to check if the module node exists in the roots directory. Thank you for sharing the code that creates them.